src/adapters/desyn/execution/src/sdk/lib.ts

```diff
@@ -0,0 +1,29 @@
+import { JsonRpcProvider } from 'ethers';
+import { UserTVLData } from "./types";
+import axios from "axios"
+
+export const getTimestampAtBlock = async (blockNumber: number) => {
+  const provider = new JsonRpcProvider("https://rpc.zklink.io");
+  const block = await provider.getBlock(blockNumber);
+  return Number(block?.timestamp);
+};
+  
+export async function getUserTransactionData(blockNumber: number):Promise<UserTVLData[]>  {
+
+  const response = (await axios.get("https://api.desyn.io/zklink/points_gateway/points/get_zklink_balance_list?blockNumber=" + blockNumber));
+  const data = response.data.list;
+
+  const userTxData: UserTVLData[] = data.map((item: any) => {
+    return {
+      blockNumber: blockNumber,
+      userAddress: item.userAddress,
+      poolAddress: item.poolAddress,
+      lpAddress: item.lpAddress,
+      tokenAddress: item.tokenAddress,
+      balance: item.balance,
```

Comment:
The `balance` needs to return the raw balance. Please ensure that the `decimals` are preserved in the returned data

Reply:
@santinoYoung Sorry for the misunderstanding. What I meant was that the `balance` needs to be returned as an integer.

Reply:
> @santinoYoung Sorry for the misunderstanding. What I meant was that the `balance` needs to be returned as an integer.

Sorry, I don't quite understand, now all that is returned is the integer and decimals precision

Reply:
For example return data 
`{
        "userAddress": "0x985925d21B12aB4a148bd794b9Af46faFD7CeE0E",
        "tokenAddress": "0xfa70fD01EbD5aa64f31E5D3575C444919C79275E",
        "lpAddress": "0xe116d8dCcA2aA97150Aa24fa4957859760AEC6fB",
        "poolAddress": "0x1ccc2353d0cC1C0B201Cf4566141188cD6F335d6",
        "balance": "1000000000000000",
        "timestamp": 1723716235,
        "decimals": 18
      }`

Reply:
@santinoYoung  excute the command `npm run adapter:tvl -- desyn tvl 5228742`
<img width="913" alt="image" src="https://github.com/user-attachments/assets/937d6e0e-dc05-4649-a2f1-fa171355d086">

Reply:
@santinoYoung run the command `npm run adapter:tvl -- desyn tvl 5260011`
<img width="1181" alt="image" src="https://github.com/user-attachments/assets/6e0ddfde-83a1-4a59-b030-d60fc81f816a">

Reply:
> npm run adapter:tvl -- desyn tvl 5260011

Oh. I got it, thanks. I'll take care of it.

Reply:
It‘s fixed it～